### ISSUE TITLE
Not saved random generated password in config for "crypt"

### ISSUE THREAD

**Issue by lithium0003** [rclone/rclone#3492]
## What is the problem you are having with rclone?
In version v1.49.0, configure crypt remote password is not working with randomly generate password, manual password is correct. In saved conf file, password obscure's IV (16 bytes) are saved but nothing followed, so it treated null password / salt. It seems be working correctly crypt remote but its password is null. In display, random generated password is shown but is not saved and gone.

## Version
v1.49.0 is not working, v1.48.0 is correctly woking.

#### version

``` bash
rclone v1.49.0
- os/arch: darwin/amd64
- go version: go1.12.9
```

#### configure
``` bash
$ rclone config
[snip.]
Password or pass phrase for encryption.
y) Yes type in my own password
g) Generate random password
n) No leave this optional password blank
y/g/n> g
Password strength in bits.
64 is just about memorable
128 is secure
1024 is the maximum
Bits> 128
Your password is: S9zb1bBPVvH35w5_mM41WA
Use this password? Please note that an obscured version of this 
password (and not the password itself) will be stored under your 
configuration file, so keep this generated password in a safe place.
y) Yes
n) No
y/n> y
Password or pass phrase for salt. Optional but recommended.
Should be different to the previous password.
y) Yes type in my own password
g) Generate random password
n) No leave this optional password blank
y/g/n> g
Password strength in bits.
64 is just about memorable
128 is secure
1024 is the maximum
Bits> 128
Your password is: 489hSmaFItYDRODJczgxsg
Use this password? Please note that an obscured version of this 
password (and not the password itself) will be stored under your 
configuration file, so keep this generated password in a safe place.
y) Yes
n) No
y/n> y
Remote config
--------------------
[crypttest]
type = crypt
remote = google:rclonetest2
filename_encryption = standard
password = *** ENCRYPTED ***
password2 = *** ENCRYPTED ***
--------------------
y) Yes this is OK
e) Edit this remote
d) Delete this remote
y/e/d> y
```

#### result
``` bash
$ cat .config/rclone/rclone.conf 
[crypttest]
type = crypt
remote = google:rclonetest2
filename_encryption = standard
password = BXBvmcEnAMMcCJEKl1Sy_w
password2 = uq75aIIZxnh8nHdD9yYt9Q
```

#### version

``` bash
rclone v1.48.0
- os/arch: darwin/amd64
- go version: go1.12.6
```

#### configure
``` bash
$ rclone config
[snip.]
Password or pass phrase for encryption.
y) Yes type in my own password
g) Generate random password
n) No leave this optional password blank
y/g/n> g
Password strength in bits.
64 is just about memorable
128 is secure
1024 is the maximum
Bits> 128
Your password is: ZFY2FeiBGTZzY-xkp9DPOQ
Use this password? Please note that an obscured version of this 
password (and not the password itself) will be stored under your 
configuration file, so keep this generated password in a safe place.
y) Yes
n) No
y/n> y
Password or pass phrase for salt. Optional but recommended.
Should be different to the previous password.
y) Yes type in my own password
g) Generate random password
n) No leave this optional password blank
y/g/n> g
Password strength in bits.
64 is just about memorable
128 is secure
1024 is the maximum
Bits> 128
Your password is: ymPHKpbHWBDvTobV-iiqJA
Use this password? Please note that an obscured version of this 
password (and not the password itself) will be stored under your 
configuration file, so keep this generated password in a safe place.
y) Yes
n) No
y/n> y
Edit advanced config? (y/n)
y) Yes
n) No
y/n> n
Remote config
--------------------
[crypttest]
type = crypt
remote = google:rclonetest2
filename_encryption = standard
password = *** ENCRYPTED ***
password2 = *** ENCRYPTED ***
--------------------
y) Yes this is OK
e) Edit this remote
d) Delete this remote
y/e/d> y
```

#### result
``` bash
$ cat .config/rclone/rclone.conf 
[crypttest]
type = crypt
remote = google:rclonetest2
filename_encryption = standard
password = 4ZLoW0upuVFuyTvxEhdil3LMOawVAHvJqF6_qSrkY8F7YL44IOo
password2 = kk0M92t7yMTseIcXWm8-uBXlBEVJyfwneYe-bJfKpjW93eyxaUc
```

**Comment by ncw** [comment]
I can confirm this - thanks for reporting.

I'll figure out what has gone wrong and issue a point release

**Comment by lithium0003** [comment]
This var password
https://github.com/rclone/rclone/blob/7211c2dca7ab0dbab0ad8f11ae50c118f098a3fd/fs/config/config.go#L856

is hidden here
https://github.com/rclone/rclone/blob/7211c2dca7ab0dbab0ad8f11ae50c118f098a3fd/fs/config/config.go#L864

**Comment by ncw** [comment]
Yes that is a nice go gotcha!  That is normally something that the linting picks up too.

I'm just testing a patch + a test and will commit in a moment.

**Comment by ncw** [comment]
I've merged a fix for this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released shortly in v1.49.1

**Comment by redronin** [comment]
Wait, did you mean the original password2 needs to be more than 22 characters or the encrypted value is more than 22. The original plain text salt is only 8 chars long. Plain text characters, no emoji or special characters.

Also not sure what 0,2,3 mod 4 means.
